### PR TITLE
Created a twisted 101 tutorial

### PR DESCRIPTION
**Pull request by qstokkink**
This should allow new recruits to get acquainted with Twisted's control flow.

### CODE REVIEW

**Comment by qstokkink** [comment]
@devos50 when you have time 👍

File: doc/twisted101.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
+Twisted in Tribler 101
```
Comment (devos50) [inline doc/twisted101.rst]: Please remove the 101, just leave the title to be `Twisted in Tribler` 👍

File: doc/twisted101.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+======================
+
+Once you have mastered basic Python, you will need to understand two more advanced concepts before you can start work on Tribler: generator functions and scheduling in Twisted.
```
Comment (devos50) [inline doc/twisted101.rst]: I would say: once you have a basic understanding of Python...

Comment (devos50) [inline doc/twisted101.rst]: ... before you can start _working_ on Tribler

File: doc/twisted101.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Generator functions
+-------------------
+When browsing the Tribler source code you will find a lot of ``yield`` statements.
+This makes this function a *generator* and as a result you will not find any ``return`` statements in this function (which would be syntactically invalid).
+The special thing about these *generators* is that they can return intermittent values, without releasing their local context.
+This is an advantage when the caller of this *generator* does not necessarily need all of the outputs the *generator* could produce.
+Instead, the caller can decide to stop iterating over the outputs of the *generator* at any given time.
+
+Take for example an identity generator:
```
Comment (devos50) [inline doc/twisted101.rst]: identity -> identifier

File: doc/twisted101.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    @inlineCallbacks
+    def some_function():
+        # Wait for some_deferred_object to be called
+        yield some_deferred_object
+        # The some_deferred_object event has happened now
+    
+This block of code does two things.
+
+1. [``yield some_deferred_object``] Yield a ``Deferred`` object
+2. [``@inlineCallbacks``] Call all values of the generator, call the ``next()`` value of this generator every time the previously yielded Deferred has been called.
```
Comment (devos50) [inline doc/twisted101.rst]: I wouldn't say a Deferred is called but instead, talk about when it fires.

**Pull request merged**